c2rust-analyze/src/main.rs

```diff
@@ -278,6 +278,35 @@ fn update_pointer_info<'tcx>(acx: &mut AnalysisCtxt<'_, 'tcx>, mir: &Body<'tcx>)
     }
 }
 
+// Walks the type `ty` and applies a function `f` to it if it's an ADT
+// `f` gets applied recursively to `ty`'s generic types and fields (if applicable)
```

Comment:
```suggestion
/// Walks the type `ty` and applies a function `f` to it if it's an ADT
/// `f` gets applied recursively to `ty`'s generic types and fields (if applicable)
```

Reply:
Could you also document what the return value of `f` means?

Reply:
Can you also add that we only look for ADTs rather than other FFI-crossing types because ADTs are the only nominal ones, which are the ones that we may rewrite?

Reply:
a2e99a57